src/darwin/Framework/CHIP/MTRLogger.h

```diff
@@ -0,0 +1,89 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef MTRLogger_h
+#define MTRLogger_h
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// MARK: - MTRLogLevel
+
+/**
+ * An enumeration of the log levels supported by the Matter SDK.
+ */
+typedef NS_ENUM(NSUInteger, MTRLogLevel) { MTRLogLevelFault = 0, MTRLogLevelError, MTRLogLevelInfo, MTRLogLevelDebug };
+
+// MARK: - MTRLogger
+
+/**
+ * An entity that can be used to log messages from the Matter SDK.
+ */
+@protocol MTRLogger
+
+/**
+ * Tests if logs may be emitted at the given log level.
+ *
+ * @param level The log level to test.
+ * @returns `true` if logs may be emitted at the given log level; `false` otherwise.
+ */
+- (BOOL)isLoggingEnabledAtLevel:(MTRLogLevel)level;
+
+/**
+ * Logs a message at the given level.
+ *
+ * @note Messages will only be passed to this method if `isLoggingEnabledAtLevel` is `true` for the given log level.
+ */
+- (void)logAtLevel:(MTRLogLevel)level message:(NSString *)message;
```

Comment:
We need to ensure that this is never called on a client-provided MTRLogger while running on the Matter dispatch queue.

That means either we need to create our own dispatch queue inside the framework for making these calls, dispatch async to it internally, and document that the API consumer cannot rely on these calls happening on any particular dispatch queue, or we need to have the framework consumer provide a dispatch queue to use when they provide an MTRLogger.

Reply:
I changed it to use the chip worker dispatch queue.